jck/build.xml

```diff
@@ -161,11 +161,34 @@
 									</then>
 								</if>
 								<echo message="Updating ${JCK_ROOT_USED} with latest..." />
-								<exec executable="git" dir="${JCK_ROOT_USED}" failonerror="true">
+								<exec executable="git" dir="${JCK_ROOT_USED}" failonerror="false" resultproperty="return.code">
 									<arg value="pull" />
 									<arg value="${JCK_GIT_REPO_USED}" />
 									<arg value="${jck_branch}" />
 								</exec>
+								<if>
+									<not>
+										<equals arg1="${return.code}" arg2="0"/>
+									</not> 
+									<then>
+										<echo message="Performing hard reset..." />
+										<exec executable="git" dir="${JCK_ROOT_USED}" failonerror="true">
+											<arg value="reset" />
+											<arg value="--hard" />
+											<arg value="origin/master" />
```

Comment:
Why we need to reset to origin/master, then clean and pull again? Could we just hard reset to ${JCK_GIT_REPO_USED} ${jck_branch} directly?

Reply:
Any idea what error we get when we do `reset --hard ${JCK_GIT_REPO_USED} ${jck_branch}`?

Reply:
Btw, here `origin  = {JCK_GIT_REPO_USED}`; e.g. Grinder_JCK/672/console. The `git clean` is just added as an extra step to clean up any possible garbage left behind after the reset.

Reply:
Did you try just `git reset --hard`?
I don't really like hardcoding origin/master since those names are not guaranteed. Especially since master is being renamed to main.

Reply:
`Did you try just git reset --hard?` - yes, and that did not get rid of the merge conflict. 

"Especially since master is being renamed to main." - do you mean in our internal JCK repos?